samples/extensions/ray_tracing_basic/ray_tracing_basic.cpp

```diff
@@ -79,46 +78,18 @@ void RaytracingBasic::request_gpu_features(vkb::PhysicalDevice &gpu)
 */
 void RaytracingBasic::create_storage_image()
 {
-	storage_image.width  = width;
-	storage_image.height = height;
-
-	VkImageCreateInfo image = vkb::initializers::image_create_info();
-	image.imageType         = VK_IMAGE_TYPE_2D;
-	image.format            = VK_FORMAT_B8G8R8A8_UNORM;
-	image.extent.width      = storage_image.width;
-	image.extent.height     = storage_image.height;
-	image.extent.depth      = 1;
-	image.mipLevels         = 1;
-	image.arrayLayers       = 1;
-	image.samples           = VK_SAMPLE_COUNT_1_BIT;
-	image.tiling            = VK_IMAGE_TILING_OPTIMAL;
-	image.usage             = VK_IMAGE_USAGE_TRANSFER_SRC_BIT | VK_IMAGE_USAGE_STORAGE_BIT;
-	image.initialLayout     = VK_IMAGE_LAYOUT_UNDEFINED;
-	VK_CHECK(vkCreateImage(get_device().get_handle(), &image, nullptr, &storage_image.image));
-
-	VkMemoryRequirements memory_requirements;
-	vkGetImageMemoryRequirements(get_device().get_handle(), storage_image.image, &memory_requirements);
-	VkMemoryAllocateInfo memory_allocate_info = vkb::initializers::memory_allocate_info();
-	memory_allocate_info.allocationSize       = memory_requirements.size;
-	memory_allocate_info.memoryTypeIndex      = get_device().get_memory_type(memory_requirements.memoryTypeBits, VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT);
-	VK_CHECK(vkAllocateMemory(get_device().get_handle(), &memory_allocate_info, nullptr, &storage_image.memory));
-	VK_CHECK(vkBindImageMemory(get_device().get_handle(), storage_image.image, storage_image.memory, 0));
-
-	VkImageViewCreateInfo color_image_view           = vkb::initializers::image_view_create_info();
-	color_image_view.viewType                        = VK_IMAGE_VIEW_TYPE_2D;
-	color_image_view.format                          = VK_FORMAT_B8G8R8A8_UNORM;
-	color_image_view.subresourceRange                = {};
-	color_image_view.subresourceRange.aspectMask     = VK_IMAGE_ASPECT_COLOR_BIT;
-	color_image_view.subresourceRange.baseMipLevel   = 0;
-	color_image_view.subresourceRange.levelCount     = 1;
-	color_image_view.subresourceRange.baseArrayLayer = 0;
-	color_image_view.subresourceRange.layerCount     = 1;
-	color_image_view.image                           = storage_image.image;
-	VK_CHECK(vkCreateImageView(get_device().get_handle(), &color_image_view, nullptr, &storage_image.view));
+	vkb::core::ImageBuilder image_builder{{width, height, 1}};
```

Comment:
I'd prefer to roll this back. The whole reason I made this (sample) so explicit, was so that people could easily adopt that code. These changes add a new layer of abstraction (on top of the framework) that would make it harder for people to follow/adopt this sample.